src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +325,37 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module, IjProjectConfig projectConfig, ProjectFilesystem projectFilesystem) {
+    Path projectRootPath = Paths.get(projectConfig.getProjectRoot());
+    Path extraModuleBasePath =
+        projectRootPath
+            .resolve("buck-out")
+            .resolve("extra-intellij-output")
```

Comment:
You can replace `enable_extra_compiler_output_modules` with a path and enable this logic when this path is not empty. No need to add two options.